pkg/ccl/backupccl/full_cluster_backup_restore_test.go

```diff
@@ -1112,3 +1112,51 @@ DROP DATABASE defaultdb;
 		{fmt.Sprint(parentID), fmt.Sprint(parentSchemaID), name, fmt.Sprint(ID)},
 	})
 }
+
+func TestFullClusterRestoreWithUserIDs(t *testing.T) {
```

Comment:
No need for you to do it since we have a lot of cleanup to do, but this one looks like a good candidate to moving to our data driven tests. I think it would look something like (in a new file in `backupccl/testdata/backup-restore/`):

```
new-server name=s1
----

exec-sql
CREATE USER test1;
CREATE USER test2;
----

query-sql
SELECT * FROM system.users ORDER BY user_id;
----
root  false 1
admin  true 2
test1 <nil> false 100
test2 <nil> false 101

exec-sql
BACKUP INTO 'nodelocal://0/test/'
----

new-server name=s2 share-io-dir=s1
----

exec-sql
RESTORE FROM LATEST IN 'nodelocal://0/test/'
----

query-sql
SELECT * FROM system.users ORDER BY user_id;
----
root  false 1
admin  true 2
test1 <nil> false 100
test2 <nil> false 101

exec-sql
CREATE USER test3
----

query-sql
SELECT * FROM system.users ORDER BY user_id;
----
root  false 1
admin  true 2
test1 <nil> false 100
test2 <nil> false 101
test3 <nil> false 102
```